### PR TITLE
use ios-deploy, remove in-house rust partial port

### PR DESCRIPTION
**Pull request by kali**

### CODE REVIEW

**Review by fredszaq — APPROVED** [review]

**Comment by kali** [comment]
Fix #166

**Comment by kali** [comment]
Fix #165.

**Comment by kali** [comment]
Fix #158

**Comment by kali** [comment]
Fix #135

File: dinghy-lib/src/ios/device.rs [hunk 1]
```diff
@@ -758,21 +378,18 @@ fn launch_app(dev: &IosSimDevice, app_args: &[&str], _envs: &[&str]) -> Result<(
...
     // We need the "exit with status" line which is the 3rd from the last
-    let lines: Vec<&str> = output
+    let exit_status_line = output
         .lines()
-        .filter(|line| line.trim().len() > 0)
         .rev()
-        .collect();
-    let exit_status_line = lines.get(1);
-    debug!("exit status line: {:?}", exit_status_line);
+        .find(|line| line.contains("exited with status"));
```
Comment (simlay) [inline dinghy-lib/src/ios/device.rs:384]: Nice. This is definitely cleaner than before.

**Review by simlay — COMMENTED** [review]
This is definitely a better way to run on an actual iOS device. It was a bit unclear to me if this removed the use of `xcrun` for running in the iOS simulator but I think not.

I think my only comment is that in the CI build I see:
```
[2022-07-26T15:29:42Z INFO  dinghy_lib::ios] Failed to execute ios-deploy to look for ios devices No such file or directory (os error 2). It this is not expected, you need to make sure `ios-deploy --json -c -t 1` works as expected.
```

When I read that, I think it's an error (even though it clearly says "INFO"). Running the command `which ios-deploy` will have an exit status of `1` if the command doesn't exist and `0` if it does. Might be a bit better than "Failed to execute" even if it's clearly a recoverable error. Thoughts?

**Comment by kali** [comment]
As far as I can tell, ios-deploy does not handle simulators, so we should keep to two paths if we want to keep the sims (and I think it's worth it since it do not breaks that often and it makes sense for CIs).

Not finding ios-deploy is definitely not an error: many people (me included) use macs for non-ios development, so we want to keep this an option. I agree that the message could be better... I can see a couple of other case where we can not execute (wrong permissions for instance) but they quite unlikely.

**Pull request merged**